### DISCUSSION TITLE
How to check classOfDevice on Airpods 3?

### DISCUSSION THREAD

**Discussion by akshilshah** [jacklandrin/OnlySwitch#42] [Q&A]
Hi,

How can I check ```classOfDevice```.  I have AirPods 3 and the Airpods switch does not work correctly.

Thanks

**Comment by jacklandrin** [comment]
Hi,
If your AirPods 3 has paired with your Mac, you can print `IOBluetoothDevice.pairedDevices()` in file `BluetoothDevicesManager` which is all paired devices with your Mac. And check the property `classOfDevice`.

**Reply by akshilshah** [reply]
Sorry, I am a noob in the world of swift programming. 

Are you suggesting I write the code in Xcode and check it or it can be done from the terminal?

Thanks

**Reply by akshilshah** [reply]
if possible can you please share with me a playground snippet?

**Reply by jacklandrin** [reply]
You can paste these codes into your playground. It should be noted that **Platform** must be *macOS* in **Playground Setting**.
```
import IOBluetooth

let devices = IOBluetoothDevice.pairedDevices()!
let allPairedDevices = devices.filter{$0 is IOBluetoothDevice}.map{$0 as! IOBluetoothDevice}
for device in allPairedDevices {
    print("\(String(describing: device.name)):\(device.classOfDevice)")
}
```

**Reply by akshilshah** [reply]
It worked thanks for your quick response. 

Output
```
Optional("AirPods Pro - Find My"):2360344
Optional("AirPods - Find My"):2360344
```

Interesting to note over here is the ```classOfDevice``` are the same for both ``AirPods pro``` and ```AirPods 3rd gen``` but it still doesn't work with the ```AirPods 3rd gen```.

Do you have any clue for me to debug?

Thanks

**Reply by jacklandrin** [reply]
I think 2360344 is not unique mark for AirPods, since some users tell me some non-airpods devices also are same class. I'm going to look for another mark in the future. 
Can it connect via system's sound icon in menu bar? Or only can't toggle airpods switch in my app?
